metrics/monitor/fd_monitor.go

```diff
@@ -31,15 +35,28 @@ func (f *FileDescriptor) Run(signals <-chan os.Signal, ready chan<- struct{}) er
 	for {
 		select {
 		case <-f.ticker.C:
-			fdInfo, err := ioutil.ReadDir(f.path)
-			if err != nil {
-				f.logger.Error("error-reading-filedescriptor-path", zap.Error(err))
-				break
+			numFds := 0
+			if runtime.GOOS == "linux" {
+				fdInfo, err := ioutil.ReadDir(f.path)
+				if err != nil {
+					f.logger.Error("error-reading-filedescriptor-path", zap.Error(err))
+					break
+				}
+				numFds = symlinks(fdInfo)
+			} else if runtime.GOOS == "darwin" {
+				// no /proc on MacOS, falling back to lsof
+				out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("lsof -p %v", os.Getpid())).Output()
+				if err != nil {
+					f.logger.Error("error-running-lsof", zap.Error(err))
+					break
+				}
+				lines := strings.Split(string(out), "\n")
+				numFds = len(lines) - 1 //cut the table header
```

Comment:
Is gorouter on macOS a supported configuration / maintained?

Reply:
this commit is actually part of #312 - not meant to be in this PR.
I don't think gorouter on macOS is supported for production use, but it should work at least for development.

Reply:
clarified on the phone, no. We'll remove it from the PR.